jupyter_rfb/_utils.py

```diff
@@ -0,0 +1,100 @@
+import io
+import builtins
+import traceback
+from base64 import encodebytes
+
+import ipywidgets
+
+from ._png import array2png
+
+_original_print = builtins.print
+
+
+class RFBOutputContext(ipywidgets.Output):
+    """An output widget with a different implementation of the context manager.
+
+    Handles prints and errors in a more reliable way, that is also
+    lightweight (i.e. no peformance cost).
+
+    See https://github.com/vispy/jupyter_rfb/issues/35
+    """
+
+    capture_print = False
+    _prev_print = None
+
+    def print(self, *args, **kwargs):
+        """Print function that show up in the output."""
+        f = io.StringIO()
+        kwargs.pop("file", None)
+        _original_print(*args, file=f, flush=True, **kwargs)
+        text = f.getvalue()
+        self.append_stdout(text)
+
+    def __enter__(self):
+        """Enter context, replace print function."""
+        if self.capture_print:
+            self._prev_print = builtins.print
+            builtins.print = self.print
+        return self
+
+    def __exit__(self, etype, value, tb):
+        """Exit context, restore print function and show any errors."""
+        if self.capture_print and self._prev_print is not None:
+            builtins.print = self._prev_print
+            self._prev_print = None
+        if etype:
+            err = "".join(traceback.format_exception(etype, value, tb))
+            self.append_stderr(err)
+            return True  # declare that we handled the exception
+
+
+class Snapshot:
```

Comment:
How does this differ from the `Image` class that I'm 99% sure exists? Could this class be replaced or subclass from that?

Here it is: https://ipywidgets.readthedocs.io/en/latest/examples/Widget%20List.html#Image

Reply:
Good point!

Reply:
I made it inherit from `IPython.display.DisplayObject` instead. We need to add the css `class`, and also have this little title node, so it felt too different from an `Image`. Anyway, by inheriting from DisplayObject we can make the docs leaner and the API cleaner :)